.github/workflows/k8s-e2e.yml

```diff
@@ -0,0 +1,155 @@
+              endpoint = ["http://${registry_ip}:5000"]
+          [plugins."io.containerd.grpc.v1.cri".registry.configs."${registry_ip}:5000".auth]
+            username = "${{ env.DOCKER_USER }}"
+            password = "${{ env.DOCKER_PASSWORD }}"
+          EOF
+          docker cp containerd.config.toml.bak kind-control-plane:/etc/containerd/config.toml.bak
+          docker exec kind-control-plane sh -c "cat /etc/containerd/config.toml.bak > /etc/containerd/config.toml"
+          docker exec kind-control-plane systemctl restart containerd
+      - name: Install Nydus binaries and convert nydus image
+        run: |
+          NYDUS_VERSION=$(curl --silent "https://api.github.com/repos/dragonflyoss/image-service/releases/latest" | grep -Po '"tag_name": "\K.*?(?=")')
+          wget -q https://github.com/dragonflyoss/image-service/releases/download/$NYDUS_VERSION/nydus-static-$NYDUS_VERSION-linux-amd64.tgz
+          tar xzf nydus-static-$NYDUS_VERSION-linux-amd64.tgz
+          sudo cp nydus-static/nydusify nydus-static/nydus-image /usr/local/bin/
+
+          registry_ip=$(ip addr show eth0 | grep 'inet ' | awk '{print $2}' | cut -d/ -f1)
+
+          sudo DOCKER_CONFIG=$HOME/.docker nydusify convert \
+                 --source busybox:latest \
+                 --target ${registry_ip}:5000/busybox:nydus-v6-latest \
+                 --fs-version 6
+      - name: Run E2E test
+        run: |
+          registry_ip=$(ip addr show eth0 | grep 'inet ' | awk '{print $2}' | cut -d/ -f1)
+          kubectl create --namespace nydus-system secret generic regcred \
+              --from-file=.dockerconfigjson=$HOME/.docker/config.json \
+              --type=kubernetes.io/dockerconfigjson
+
+          sed -e "s|REGISTRY_IP|${registry_ip}|" tests/e2e/k8s/test-pod.yaml.tpl > tests/e2e/k8s/test-pod.yaml
+
+          kubectl apply -f tests/e2e/k8s/test-pod.yaml
+          kubectl wait po test-pod --for=condition=ready --timeout=1m
+          kubectl delete -f tests/e2e/k8s/test-pod.yaml
+      - name: Dump logs
+        if: failure()
+        continue-on-error: true
+        run: |
+          log_dir="/tmp/nydus-log"
+          mkdir -p $log_dir
+          export ns=nydus-system
+          for p in `kubectl -n $ns get pods --no-headers -o custom-columns=NAME:metadata.name`; do
+           kubectl -n $ns get pod $p -o yaml >> $log_dir/nydus-pods.conf
+           kubectl -n $ns describe pod $p  >> $log_dir/nydus-pods.conf
+           kubectl -n $ns logs $p -c nydus-snapshotter >> $log_dir/nydus-snapshotter.log || echo "failed to get snapshotter log"
+          done
+          kubectl -n $ns get secrets -o yaml >> $log_dir/nydus-secrets.log
+
+          docker exec kind-control-plane cat /etc/containerd/config.toml >> $log_dir/containerd-config.toml
+          docker exec kind-control-plane containerd config dump >> $log_dir/containerd-config-dump.toml
+
+          docker exec kind-control-plane journalctl -u containerd >> $log_dir/containerd.log
```

Comment:
Do we need to append `--no-pager`, otherwise I am not sure if the containerd log can be piped

Reply:
Added.
But from the test result, the log wasn't truncated, since there is no tty.